### ISSUE TITLE
ScanCode version 31.1.0 produces `license_expression` in `packages` with duplicated licenses

### ISSUE THREAD

**Issue by francescayeye** [nexB/scancode-toolkit#3073]
### Description

> Please leave a brief description of the bug or feature request:

ScanCode version 31.1.0 produces `license_expression` in `packages` with duplicated licenses

### How To Reproduce

> Tell us how to reproduce the issue. 

```
$ scancode --version
ScanCode version: 31.1.0
ScanCode Output Format version: 2.0.0
SPDX License list version: 3.17

$ pip install --isolated -t /tmp/double-license cffi==1.15.1
Collecting cffi==1.15.1
  Using cached cffi-1.15.1-cp39-cp39-macosx_10_9_x86_64.whl (179 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Installing collected packages: pycparser, cffi
Successfully installed cffi-1.15.1 pycparser-2.21


$ scancode -lp --json-pp double-license.json /tmp/double-license
Setup plugins...
Collect file inventory...
Scan files for: licenses, packages with 1 process(es)...
[####################] 172                                          
Scanning done.
Summary:        licenses, packages with 1 process(es)
Errors count:   0
Scan Speed:     1.94 files/sec. 
Initial counts: 95 resource(s): 86 file(s) and 9 directorie(s) 
Final counts:   95 resource(s): 86 file(s) and 9 directorie(s) 
Timings:
  scan_start: 2022-08-31T092234.606571
  scan_end:   2022-08-31T092320.893715
  setup_scan:licenses: 1.86s
  setup: 1.86s
  scan: 44.37s
  total: 46.31s
Removing temporary files...done.

$ jq '.packages[] | select(.name == "cffi") | .license_expression ' double-license.json 
"mit AND mit"
```

### System configuration

> For bug reports, it really helps us to know:

* What OS are you running on? (Windows/MacOS/Linux) MacOs 12.5.1 (21G83)
* What version of scancode-toolkit was used to generate the scan file? 31.1.0
* What installation method was used to install/run scancode? (pip/source download/other) pip

**Comment by pombredanne** [comment]
Thanks!
@DennisClark @chinyeungli @mjherzog @JonoYang @AyanSinhaMahapatra ping...
The key question is whether or not we should simplify these expressions and remove duplicates (the remove is easy since @chinyeungli  did contribute code to this effect in license-expression)
@AyanSinhaMahapatra this could also have some impect on your pending PR on license detection

**Comment by DennisClark** [comment]
@pombredanne    we should definitely simplify these license expressions by removing duplicates when possible.

**Comment by AyanSinhaMahapatra** [comment]
This is fixed in https://github.com/nexB/scancode-toolkit/tree/add-license-detection

I.e. we simplify these expressions and this is not `mit AND mit` anymore, but just `mit`.

Attaching a scan of the same from the branch I linked above for reference, notice the two seperate License Detections and the `declared_license_expression` is `mit`:
[double-license.json.txt](https://github.com/nexB/scancode-toolkit/files/9462637/double-license.json.txt)

**Comment by francescayeye** [comment]
great @AyanSinhaMahapatra 

is `declared_license_expression` a new field or renaming of `license_expression` ?

**Comment by AyanSinhaMahapatra** [comment]
@aspacca yes! We have a couple of these renaming and a lot of other output format changes w.r.t license detection with a lot of other related improvements. 

See #2065 for more info on the renaming.

**Comment by francescayeye** [comment]
thanks for clarifying @AyanSinhaMahapatra 
what version should I pin as higher dependency in order to be sure not have break because of the renaming and output format changes? thanks

**Comment by AyanSinhaMahapatra** [comment]
@aspacca we really don't recommend keeping pinned to a dependency for too long as you'll miss out on a lot of important license updates, bugfixes and new features. We have a ouput version changes documentation at https://scancode-toolkit-reference-scans.readthedocs.io/en/latest/ with diffs and explanations for each changes in output format version. You can refer to this for the magnitude of changes and plan what you have to do to upgrade.

[v21.8.4 -> v30.1.0](https://scancode-toolkit-reference-scans.readthedocs.io/en/latest/outputs/latest/clipeu-v30.1.0-v21.8.4-1a0bd63.html) : Didn't have that many major changes.

[30.x.x ->  31.x.x](https://scancode-toolkit-reference-scans.readthedocs.io/en/latest/outputs/latest/clipeu-v31.0.0b4-v30.1.0-37d4e0a.html) : Has major changes in package data, but some is renaming and some are codebase-level additions so this can be easily upgraded to,

`31.x.x ->  32.x.x` would have major changes in license detection data.
